python/ray/data/preprocessors/batch_mapper.py

```diff
@@ -36,19 +39,56 @@ class BatchMapper(Preprocessor):
         >>> preprocessor = BatchMapper(fn)
         >>> preprocessor.transform(ds)  # doctest: +SKIP
         Dataset(num_blocks=1, num_rows=3, schema={X: int64})
+        >>>
+        >>> def fn_numpy(batch: Dict[str, np.ndarray]) -> Dict[str, np.ndarray]:
+        ...     return {"X": batch["X"]}
+        >>> preprocessor = BatchMapper(fn_numpy, batch_format="numpy")
+        >>> preprocessor.transform(ds)  # doctest: +SKIP
+        Dataset(num_blocks=1, num_rows=3, schema={X: int64})
 
     Args:
         fn: The function to apply to data batches.
+        batch_format: The preferred batch format to use in UDF. If not given,
+            we will infer based on the input dataset data format.
     """
 
     _is_fittable = False
 
-    def __init__(self, fn: Callable[["pandas.DataFrame"], "pandas.DataFrame"]):
+    def __init__(
+        self,
+        fn: Union[
+            Callable[["pandas.DataFrame"], "pandas.DataFrame"],
+            Callable[
+                [Union[np.ndarray, Dict[str, np.ndarray]]],
+                Union[np.ndarray, Dict[str, np.ndarray]],
+            ],
+        ],
+        batch_format: Optional[str] = None,
```

Comment:
Should we make the default `batch_format="pandas"` for the sake of backwards compatibility? IIUC this will now delegate to `_transform_numpy` and will provide NumPy batches to the UDF for Arrow datasets (e.g. post-read tabular datasets: `ds = ray.data.read_parquet()`) when previously it would provide pandas DataFrame batches to the UDF.

Or are we fine with this breaking change?

Reply:
I initially had it as `pandas` as well but later realized actually most of existing tests + examples are ok. Only place i need to update BatchMapper is the hugging face example which supposely be using numpy UDF given it's handling tokenizer